src/metrics.py

```diff
@@ -0,0 +1,66 @@
+#  metrics and losses
+
+import torch
+
+def precision(pred, true, label=1):
+    tp = ((pred == label) & (true == label)).sum().item()
+    fp = ((pred == label) & (true != label)).sum().item()
+
+    if tp == fp == 0:
+        return 0
+
+    return tp / (tp + fp)
+
+def tp_fp_fn(pred, true, label=1):
+    tp = ((pred == label) & (true == label)).sum().item()
+    fp = ((pred == label) & (true != label)).sum().item()
+    fn = ((pred != label) & (true == label)).sum().item()
+
+    return tp, fp, fn
+
+
+def recall(pred, true, label=1):
+    tp = ((pred == label) & (true == label)).sum().item()
+    fp = ((pred == label) & (true != label)).sum().item()
+    fn = ((pred != label) & (true == label)).sum().item()
+
+    return tp / (tp + fn)
+
+
+def pixel_acc(pred, true):
+    return (pred == true).sum().item() / true.numel()
```

Comment:
is .sum() / .numel() not the same as .mean()?

Reply:
I'd have to transform from bool to float then take the mean. Either way should work.